### PR TITLE
Upgrade to React v15.0.0

### PR DESCRIPTION
**Pull request by nathanielparke**
Upgrades react by removing the LinkedStateMixin

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -13,10 +13,10 @@
     "jquery": "2.1.1",
     "jquery-ui": "1.10.5",
     "js-cookie": "~2.0.3",
-    "react": "0.14.8",
-    "react-dom": "0.14.8",
+    "react": "15.0.0",
+    "react-dom": "15.0.0",
     "react-router": "1.0.0",
-    "react-addons-linked-state-mixin": "0.14.8"
```
Comment (kmeht) [inline package.json]: We seemed to remove this everywhere, do we still need the dependency?

File: huxley/www/static/js/huxley/components/TextInput.js [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+        type="text"
+        placeholder={this.props.placeholder}
+        onChange={this._handleChange}
+        value={this.props.value}
+      />
+    );
+  },
+
+  _handleChange: function(event) {
+    var value = event.target.value.replace(/[^\D]/, '');
```
Comment (kmeht) [inline huxley/www/static/js/huxley/components/TextInput.js]: I don't think we should strip numbers. Text can certainly contain numbers, like addresses.

**Comment by kmeht** [comment]
What's the criteria for deciding to use a custom change handler vs `_handleRawInputChange`?

**Comment by tdowds** [comment]
I might have missed this somewhere, but is there a reason why we're removing LinkedStateMixin?

File: package.json (hunk 1 above)
Comment (nathanielparke) [inline package.json]: Nope we can remove it. Will do that!

**Comment by nathanielparke** [comment]
@kmeht Criteria was that _handleRawInputChange was the handler that is passed into our custom components such as NumberInput and TextInput. TextInput couldn't be used with things like addresses, passwords and usernames so the addition of those extra custom handlers was necessary. I think using _handleRawInputChange for all of them is a better idea though. Cleaner and matches linkedStateMixin functionality more closely.

**Comment by kmeht** [comment]
@nathanielparke, you'll probably want to rebase on top of @m-j-mcdonald's latest diff, as there were some overlapping features.

**Pull request merged**